idaes/ui/fsvis/static/js/stream_table.js

```diff
@@ -0,0 +1,118 @@
+export class StreamTable {
+    constructor(app, model) {
+        this._app = app;
+        this.initTable(model);
+    };
+
+    initTable(model) {
+        // Clear the table first in case this is a call on refresh then call all of the methods to fill the table and setup the events
+        this.clearTable();
+        this.fillTable(model);
+        this.setupEvents();
+    }
+
+    clearTable() {
+        // Clear the table
+        $("#hide-fields-list").empty();
+        console.log(document.querySelector("#stream-table-data"))
+        $("#stream-table-data").empty();
+    }
+
+    fillTable(model) {
+        // This method fills the table with the streams and information from the model
+
+        // Get the stream table data from the html
+        let stream_table_data = model["model"]["stream_table"];
+
+        // Get the hide fields list
+        const hide_fields_list = document.querySelector("#hide-fields-list");
+
+        // Specify the column headers
+        let columns = stream_table_data["columns"];
+        let column_defs = [];
+        for (let col in columns) {
+            // There is an empty column because of the way that the pandas dataframe was oriented so 
+            // only add the columns that don't have an empty column header
+            if (columns[col] !== "") {
+                column_defs.push({headerName: columns[col], field: columns[col], filter: 'agTextColumnFilter', sortable: true});
+                let list_item = document.createElement("li");
+                let checkbox_item = document.createElement("div");
+                checkbox_item.class = "checkbox";
+                // checkbox_item.id = columns[col] + "-checkbox"
+                checkbox_item.innerHTML = '<label><input type="checkbox" value="' + columns[col] + '" id="' + columns[col] + '" checked>' + columns[col] + '</label>';
+                list_item.appendChild(checkbox_item);
+                hide_fields_list.appendChild(list_item);
```

Comment:
Since it's used a few times in the loop, you could consider extracting `columns[col]` to a variable. This would also come in handy for e.g. using a more compact string template to create `checkbox_item.innerHTML`.

Reply:
Good catch. Thanks!